test/images/webhook/main.go

```diff
@@ -43,46 +44,56 @@ func (c *Config) addFlags() {
 		"File containing the default x509 private key matching --tls-cert-file.")
 }
 
+func toAdmissionReviewStatus(err error) *v1alpha1.AdmissionReviewStatus {
+	return &v1alpha1.AdmissionReviewStatus{
+		Result: &metav1.Status{
+			Message: err.Error(),
+		},
+	}
+}
+
 // only allow pods to pull images from specific registry.
 func admit(data []byte) *v1alpha1.AdmissionReviewStatus {
 	ar := v1alpha1.AdmissionReview{}
-	if err := json.Unmarshal(data, &ar); err != nil {
+	deserializer := codecs.UniversalDeserializer()
+	if _, _, err := deserializer.Decode(data, nil, &ar); err != nil {
 		glog.Error(err)
-		return nil
+		return toAdmissionReviewStatus(err)
 	}
 	podResource := metav1.GroupVersionResource{Group: "", Version: "v1", Resource: "pods"}
```

Comment:
corev1.Resource("pods")

Reply:
Done.

Reply:
Actually can't do this yet, the types don't match: metav1.gvk v.s. schema.gvk.